### DISCUSSION TITLE
Can't open websocket connection, FastAPI keeps trying to use http

### DISCUSSION THREAD

**Discussion by nakajimayoshi** [fastapi/fastapi#11621] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import WebSocket
@staticmethod
@app.websocket("/ping")
async def authenticate_device(websocket: WebSocket):
    await websocket.accept()
    while True:
        data = await websocket.receive_text()
        await websocket.send_text(f"Message text was: {data}")
```

### Description

I am using the cli program [websocat](https://github.com/vi/websocat) to quickly test out adding a new websocket feature to an api, but every time I attempt to connect the client throws a ~~404 not found~~ I/O failure error and the server indicates it's attempting an http connection:

```
2024-05-22 10:08:35 WARNING:  No supported WebSocket library detected. Please use "pip install 'uvicorn[standard]'", or install 'websockets' or 'wsproto' manually.
2024-05-22 10:08:35 2024-05-22 01:08:35,176 317 [INFO]: BGN CALL GET http://127.0.0.1/v1/auth/device FROM 127.0.0.1
2024-05-22 10:08:35 2024-05-22 01:08:35,177 317 [INFO]: END CALL GET http://127.0.0.1/v1/auth/device FROM 127.0.0.1 PROCESS-TIME 1.279ms
2024-05-22 10:08:35 INFO:     127.0.0.1- "GET /ping HTTP/1.1" 404 Not Found
```

this is the result of typing:

```bash
websocat ws://127.0.0.1:8080/ping
```

in the terminal. 

running a GET request with the same api runs just fine. 

As for this warning  

```
2024-05-22 10:08:35 WARNING:  No supported WebSocket library detected. Please use "pip install 'uvicorn[standard]'", or install 'websockets' or 'wsproto' manually.
```

I already ran `pip install 'uvicorn[standard]'` in the server terminal and it's told me it already has the websockets package installed, so I don't know why it keeps showing. 

### Operating System

Linux

### Operating System Details

Server: Linux, Client: macOS

### FastAPI Version

0.110.0

### Pydantic Version

2.6.4

### Python Version

3.9.16

### Additional Context

running the server behind an nginx proxy with the following config:

```.conf
user  nginx;
worker_processes  auto;

error_log  /var/log/nginx/error.log notice;
pid        /var/run/nginx.pid;

events {
    worker_connections  1024;
}

http {
    include       /etc/nginx/mime.types;
    default_type  application/octet-stream;

    log_format  main  '$remote_addr - $remote_user [$time_local] "$request" '
                      '$status $body_bytes_sent "$http_referer" '
                      '"$http_user_agent" "$http_x_forwarded_for"';

    access_log  /var/log/nginx/access.log  main;
    error_log /var/log/nginx/error.log;

    sendfile        on;
    #tcp_nopush     on;
    keepalive_timeout  65;

    gzip  on;

    server {
        listen 80;       # IPv4
        listen [::]:80;  # IPv6

        proxy_set_header X-Forwarded-For   $host;        
        proxy_set_header X-Forwarded-Proto $scheme;     
        proxy_set_header Request-Path      $request_uri; 

        location / {
            proxy_pass http://app:8080/;
            proxy_http_version 1.1;
            proxy_set_header Upgrade $http_upgrade;
            proxy_set_header Connection "upgrade";
            proxy_set_header Host $host;
        }
}
```

**Comment by YuriiMotov** [comment, marked as answer]
Make sure that you run `pip install 'uvicorn[standard]'` in the right enviroument.
Looks like you still don't have Websocket library installed in the enviroument where you run your server.
Activate enviroument in the terminal before installing library and make sure you run server in the same enviroument
